akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1927,3 +1928,152 @@ private[stream] object Collect {
   override def toString = "StatefulMapConcat"
 
 }
+
+/**
+ * INTERNAL API
+ */
+@InternalApi final private[akka] class LazyFlow[I, O, M](flowFactory: I ⇒ Future[Flow[I, O, M]], zeroMat: () ⇒ M)
+  extends GraphStageWithMaterializedValue[FlowShape[I, O], M] {
+  val in = Inlet[I]("lazyFlow.in")
+  val out = Outlet[O]("lazyFlow.out")
+  override def initialAttributes = DefaultAttributes.lazyFlow
+  override val shape: FlowShape[I, O] = FlowShape.of(in, out)
+
+  override def toString: String = "LazyFlow"
+
+  override def createLogicAndMaterializedValue(inheritedAttributes: Attributes) = {
+    lazy val decider = inheritedAttributes.mandatoryAttribute[SupervisionStrategy].decider
+
+    var completed = false
+    var matVal: Option[M] = None
+    val stageLogic = new GraphStageLogic(shape) with InHandler with OutHandler {
+      val subSink = new SubSinkInlet[O]("LazyFlowSubSink")
+      override def preStart(): Unit = pull(in)
+
+      override def onPush(): Unit = {
+        try {
+          val element = grab(in)
+          val cb: AsyncCallback[Try[Flow[I, O, M]]] =
+            getAsyncCallback {
+              case Success(flow) ⇒ initInternalSource(flow, element)
+              case Failure(e)    ⇒ failure(e)
+            }
+          flowFactory(element).onComplete { cb.invoke }(ExecutionContexts.sameThreadExecutionContext)
+          setHandler(in, new InHandler {
+            override def onPush(): Unit = ()
```

Comment:
is it possible that this will be called?

Reply:
No, it is not possible. Those handlers are only for Failure / Finish handling for the time between this very moment and the moment when the `flowFactory(element)` completes and installs another set of handlers. There is no matching `pull(in)` that could trigger this `push`.

Reply:
then throw `IllegalStateException` so that we would detect bug if any

Reply:
the comments are in wrong order. but answering @ktoso 's question, it is not possible because this element is handled by the `onPush` inside which this handler change happens. I will add `InternalStateException` here.